openml/setups/setup.py

```diff
@@ -7,6 +7,7 @@
 import openml.flows
 
 
+# TODO(eddiebergman): All functionality handled by dataclass
```

Comment:
```suggestion
```

Reply:
We can open an issue for this if desired. But dataclasses don't handle type validation, have a different representation, so in the end you would have most of this anyway. Pydantic may be a nicer way to handle some of this, but then we would add an extra dependency.